### ISSUE TITLE
DOM Access in Web Workers

### ISSUE THREAD

**Issue by steelbrain** [atom/electron#1490]
Just wondering how much difficult it would be to support DOM access from Web Workers. and looking at the benefits, Atom Editor is gonna fly if we do that.

**Comment by YuriSolovyov** [comment]
Workers are not for that, besides, IPC would make it slow enough to cancel all benefits.

Something you can try is to ask Worker to pre-render some components to string, but I'm not sure if it worth it.

**Comment by steelbrain** [comment]
I didn't know IPC is `that` slow, is there any benchmarks or something?

**Comment by mark-hahn** [comment]
Processes give you all the same benefits of web workers and more.  Also the
same problem with ipc speed and lack of dom access.

Web workers are for use in a real browser that can't access processes.
They can't access the dom because the dom is not thread-safe.

**Comment by mark-hahn** [comment]
> I didn't know IPC is that slow

It is the speed of serializing objects to text and back again.  Slow is a
relative term.

**Comment by paulcbetts** [comment]
@steelbrain The problem with IPC isn't the actual transport of it being slow, is that it tends to block up processes, especially sync IPC. If the renderer process is parked waiting on a browser process to respond, or browser is trying to send IPC to renderer but renderer is busy doing other stuff, you end up with hard-to-debug delays. The less IPC you can get away with, the better.

**Comment by steelbrain** [comment]
Understandable. Thanks.
